File: Java/src/Sqrt(x).java
public class Solution
{
  public int sqrt(int x)
  {
    if (x == 0 || x == 1)
      return x;
    int left = 1;
    int right = x;
    int ans = 0;

    while (left <= right)
    {
      int mid = (left + right) / 2;
      if (mid < x / mid)
      {
        left = mid + 1;
        ans = mid;
      }
      else if (mid > x / mid)
      {
        right = mid - 1;

      }
      else
      {
        return mid;
      }
    }
    return ans;
  }
}

File: Python/Sqrt(x).py
class Solution:
    # @param x, an integer
    # @return an integer
    def sqrt(self, x):
        ans = 0
        bit = 1 << 15
        while bit > 0:
            ans |= bit
            if ans * ans > x:
                ans ^=bit
            bit >>=1
        return int(ans)